src/libbson/tests/test-bson-cmp.c

```diff
@@ -0,0 +1,309 @@
+   BSON_ASSERT (bson_cmp_less_equal_su (-1, 0u));
+   BSON_ASSERT (bson_cmp_less_equal_su (-1, 1u));
+   BSON_ASSERT (!bson_cmp_less_equal_su (1, 0u));
+   BSON_ASSERT (bson_cmp_less_equal_su (1, 1u));
+
+   BSON_ASSERT (bson_cmp_less_equal_us (0u, 0));
+   BSON_ASSERT (!bson_cmp_less_equal_us (0u, -1));
+   BSON_ASSERT (bson_cmp_less_equal_us (0u, 1));
+   BSON_ASSERT (!bson_cmp_less_equal_us (1u, 0));
+   BSON_ASSERT (!bson_cmp_less_equal_us (1u, -1));
+   BSON_ASSERT (bson_cmp_less_equal_us (1u, 1));
+}
+
+static void
+test_bson_cmp_greater_equal (void)
+{
+   BSON_ASSERT (bson_cmp_greater_equal_ss (0, 0));
+   BSON_ASSERT (bson_cmp_greater_equal_ss (0, -1));
+   BSON_ASSERT (!bson_cmp_greater_equal_ss (0, 1));
+   BSON_ASSERT (!bson_cmp_greater_equal_ss (-1, 0));
+   BSON_ASSERT (bson_cmp_greater_equal_ss (-1, -1));
+   BSON_ASSERT (!bson_cmp_greater_equal_ss (-1, 1));
+   BSON_ASSERT (bson_cmp_greater_equal_ss (1, 0));
+   BSON_ASSERT (bson_cmp_greater_equal_ss (1, -1));
+   BSON_ASSERT (bson_cmp_greater_equal_ss (1, 1));
+
+   BSON_ASSERT (bson_cmp_greater_equal_uu (0u, 0u));
+   BSON_ASSERT (!bson_cmp_greater_equal_uu (0u, 1u));
+   BSON_ASSERT (bson_cmp_greater_equal_uu (1u, 0u));
+   BSON_ASSERT (bson_cmp_greater_equal_uu (1u, 1u));
+
+   BSON_ASSERT (bson_cmp_greater_equal_su (0, 0u));
+   BSON_ASSERT (!bson_cmp_greater_equal_su (0, 1u));
+   BSON_ASSERT (!bson_cmp_greater_equal_su (-1, 0u));
+   BSON_ASSERT (!bson_cmp_greater_equal_su (-1, 1u));
+   BSON_ASSERT (bson_cmp_greater_equal_su (1, 0u));
+   BSON_ASSERT (bson_cmp_greater_equal_su (1, 1u));
+
+   BSON_ASSERT (bson_cmp_greater_equal_us (0u, 0));
+   BSON_ASSERT (bson_cmp_greater_equal_us (0u, -1));
+   BSON_ASSERT (!bson_cmp_greater_equal_us (0u, 1));
+   BSON_ASSERT (bson_cmp_greater_equal_us (1u, 0));
+   BSON_ASSERT (bson_cmp_greater_equal_us (1u, -1));
+   BSON_ASSERT (bson_cmp_greater_equal_us (1u, 1));
+}
+
+/* Sanity check: ensure ssize_t limits are as expected relative to size_t. */
+BSON_STATIC_ASSERT2 (ssize_t_size_min_check, SSIZE_MIN + 1 == -SSIZE_MAX);
+BSON_STATIC_ASSERT2 (ssize_t_size_max_check, (size_t) SSIZE_MAX <= SIZE_MAX);
+
+static void
+test_bson_in_range (void)
+{
+   const int64_t int8_min = INT8_MIN;
+   const int64_t int8_max = INT8_MAX;
+   const int64_t int32_min = INT32_MIN;
+   const int64_t int32_max = INT32_MAX;
+
+   const uint64_t uint8_max = UINT8_MAX;
+   const uint64_t uint32_max = UINT32_MAX;
+
+   const ssize_t ssize_min = SSIZE_MIN;
+   const ssize_t ssize_max = SSIZE_MAX;
+
+   BSON_ASSERT (!bson_in_range_signed (int8_t, int8_min - 1));
+   BSON_ASSERT (bson_in_range_signed (int8_t, int8_min));
+   BSON_ASSERT (bson_in_range_signed (int8_t, 0));
+   BSON_ASSERT (bson_in_range_signed (int8_t, int8_max));
+   BSON_ASSERT (!bson_in_range_signed (int8_t, int8_max + 1));
+
+   BSON_ASSERT (bson_in_range_unsigned (int8_t, 0u));
+   BSON_ASSERT (bson_in_range_unsigned (int8_t, (uint64_t) int8_max));
```

Comment:
Suggest removing unnecessary cast to `uint64_t` for consistency with other calls. Here and below.